website/docs/r/dev_test_linux_virtual_machine.html.markdown

```diff
@@ -0,0 +1,156 @@
+    offer     = "UbuntuServer"
+    publisher = "Canonical"
+    sku       = "18.04-LTS"
+    version   = "latest"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Dev Test Machine. Changing this forces a new resource to be created.
+
+-> **NOTE:** The validation requirements for the Name change based on the `os_type` used in this Virtual Machine. For a Linux VM the name must be between 1-62 characters, and for a Windows VM the name must be between 1-15 characters. It must begin and end with a letter or number, and cannot be all numbers.
+
+* `lab_name` - (Required) Specifies the name of the Dev Test Lab in which the Virtual Machine should be created. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the resource group in which the Dev Test Lab resource exists. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the Dev Test Lab exists. Changing this forces a new resource to be created.
+
+* `gallery_image_reference` - (Required) A `gallery_image_reference` block as defined below.
+
+* `lab_subnet_name` - (Required) The name of a Subnet within the Dev Test Virtual Network where this machine should exist. Changing this forces a new resource to be created.
+
+* `lab_virtual_network_id` - (Required) The ID of the Dev Test Virtual Network where this Virtual Machine should be created. Changing this forces a new resource to be created.
+
+* `size` - (Required) The Machine Size to use for this Virtual Machine, such as `Standard_F2`. Changing this forces a new resource to be created.
+
+* `storage_type` - (Required) The type of Storage to use on this Virtual Machine. Possible values are `Standard` and `Premium`.
+
+* `username` - (Required) The Username associated with the local administrator on this Virtual Machine. Changing this forces a new resource to be created.
+
+---
+
+* `allow_claim` - (Optional) Can this Virtual Machine be claimed by users? Defaults to `true`.
+
+* `disallow_public_ip_address` - (Optional) Should the Virtual Machine be created without a Public IP Address? Changing this forces a new resource to be created.
+
+* `inbound_nat_rule` - (Optional) One or more `inbound_nat_rule` blocks as defined below. Changing this forces a new resource to be created.
+
+-> **NOTE:** If any `inbound_nat_rule` blocks are specified then `disallow_public_ip_address` must be set to `true`.
+
+* `notes` - (Optional) Any notes about the Virtual Machine.
+
+* `password` - (Optional) The Password associated with the `username` used to login to this Virtual Machine. Changing this forces a new resource to be created.
+
+* `ssh_key` - (Optional) The SSH Key associated with the `username` used to login to this Virtual Machine. Changing this forces a new resource to be created.
+
+-> **NOTE:** One or either `password` or `ssh_key` must be specified.
```

Comment:
Can you specify both? If not should we add `ConflictsWith` into the schema?

Reply:
AFAIK yes - VM's allow specifying the password and ssh key